QgisModelBaker/libqgsprojectgen/dbconnector/db_connector.py

```diff
@@ -159,6 +175,42 @@ def get_bags_of_info(self):
         '''
         return []
 
+    def get_ignored_layers(self):
+        '''
+        The ignored layers according to the ignored schemas and ignored tables and the ignored ili elements
+        listed in the config.py.
+        Additionally all the ili elements that have the attribute name ili2db.mapping in the meta attribute
+        table.
+        '''
+        tables_info = self.get_tables_info()
+        relations_info = self.get_relations_info()
+        meta_attrs_info = self.get_meta_attrs_info()
+        mapping_ili_elements = []
+        tables = []
+        referencing_tables = []
+        for record in meta_attrs_info:
+            if record['attr_name'] == 'ili2db.mapping':
+                mapping_ili_elements.append(record['ilielement'])
+        for record in tables_info:
+            if 'ili_name' in record:
+                if record['ili_name'] in mapping_ili_elements or record['ili_name'] in IGNORED_ILI_ELEMENTS:
+                    tables.append(record['tablename'])
+                    continue
+            if 'schemaname' in record:
+                if record['schemaname'] in IGNORED_SCHEMAS:
+                    tables.append(record['tablename'])
+                    continue
+            if 'tablename' in record:
+                if record['tablename'] in IGNORED_TABLES:
+                    tables.append(record['tablename'])
+                    continue
+        # get the referencing tables
+        for record in relations_info:
+            if record['referenced_table'] in tables:
+                referencing_tables.append(record['referencing_table'])
```

Comment:
This `if` block is damaging our projects.

If we have basket cols in our DB, we end up with relations like this:

`{'referencing_table': 'terreno', 'referencing_column': 'T_basket', 'referenced_table': 'T_ILI2DB_BASKET', 'referenced_column': 'T_Id', 'constraint_name': 'terreno_T_basket_T_ILI2DB_BASKET_T_Id', 'strength': ''}`

Which then enter into such `if` block, and then ALL (yes, all! :smile:) our classes that are not domains, are added to the referencing_tables list, and therefore, are ignored.

Please give such if clause a second thought.

Reply:
Is `T_ILI2DB_BASKET` table contained in the `t_ili2db_meta_attrs` as `mapping`?

So  shouldn't the referencing tables be ignored at all? Or what is the criteria to ignore it or not?

Reply:
> Is T_ILI2DB_BASKET table contained in the t_ili2db_meta_attrs as mapping?

Nope, it doesn't appear there.

`T_ILI2DB_BASKET`is in the `DBConnector.config.IGNORED_TABLES` list.

The criterion here is to avoid loading such table (this works fine at the moment), but all tables that are referencing to `T_ILI2DB_BASKET` should still be loaded.

Reply:
What I see is that the `if` condition is too broad, as it also takes into account the `DBConnector.config.IGNORED_TABLES`. 

From what I understand, you want to filter out tables contained in the `t_ili2db_meta_attrs` as `mapping` and their referencing tables. Right?

Reply:
Ah I see. I did it on purpose. But obviously wrong. So I'll change that. Thanks.